File: tests/fixtures/maven-sample/src/test/java/sample/CalcTest.java
package sample;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class CalcTest {
    private final Calc calc = new Calc();

    @Test
    void testAdd() {
        assertEquals(5, calc.add(2, 3));
    }

    @Test
    void testIsEven() {
        assertTrue(calc.isEven(4));
        assertFalse(calc.isEven(7));
    }

    @Test
    void testForceFailable() {
        // -DforceFail=true 让此用例失败，用于 Spike-2 Red 验收
        if ("true".equals(System.getProperty("forceFail"))) {
            fail("forced failure via -DforceFail=true (Spike-2 Red 验收)");
        }
    }
}

File: tests/fixtures/jest-sample/calc.test.js
// Spike-3 fixture：Jest 测试用例（含 PASS + 可控 FAIL）。
// AISE_FIXTURE_FORCE_FAIL=1 让 forceFailable 失败，模拟 Red 状态。
const { add, isEven } = require('./calc');

describe('calc', () => {
  test('add', () => {
    expect(add(2, 3)).toBe(5);
  });

  test('isEven', () => {
    expect(isEven(4)).toBe(true);
    expect(isEven(7)).toBe(false);
  });

  test('forceFailable', () => {
    if (process.env.AISE_FIXTURE_FORCE_FAIL === '1') {
      throw new Error('forced failure via AISE_FIXTURE_FORCE_FAIL=1 (Spike-3 Red 验收)');
    }
  });
});
